iot/iot-tenant-cleaner/src/main/java/io/enmasse/iot/tenantCleaner/infinispan/Application.java

```diff
@@ -0,0 +1,182 @@
+            System.err.println(String.format("[ERROR] Unable to access Infinispan caches. {}", e.getMessage()));
+            System.exit(1);
+        }
+
+        // Query and delete entries in devicesCache
+        QueryFactory queryFactory = Search.getQueryFactory(devicesCache);
+        Query query = queryFactory.from(DeviceInformation.class)
+                .select(Expression.property("deviceId"))
+                .having("tenantId")
+                .eq(TENANT_ID_TO_CLEAN)
+                .build();
+
+        List<String[]> matches = query.list();
+        matches.forEach(deviceInfo -> {
+            System.out.println(deviceInfo[0]);
+            DeviceKey key = new DeviceKey(TENANT_ID_TO_CLEAN, deviceInfo[0]);
+            devicesCache.removeAsync(key);
+
+            DeviceConnectionKey conKey = new DeviceConnectionKey(TENANT_ID_TO_CLEAN, deviceInfo[0]);
+            devicesConnectionCache.removeAsync(conKey);
+        });
+
+        // Stop the cache managers and release all resources
+        try {
+            deviceconProvider.stop();
+            mgmtProvider.stop();
+        } catch (Exception n){
+            System.err.println("[WARNING] Could not properly release cache manager resources in infinispan");
+        }
+    }
+
+    private void configure(){
+        ConfigStoreOptions yaml = new ConfigStoreOptions()
+                .setType("file")
+                .setFormat("yaml")
+                .setConfig(new JsonObject()
+                        .put("path", "/resources/config.yaml")
+                );
+
+        ConfigStoreOptions env = new ConfigStoreOptions()
+                .setType("env");
+
+        ConfigRetriever retriever = ConfigRetriever.create(vertx,
+                new ConfigRetrieverOptions().addStore(yaml).addStore(env));
+
+        Future configured = Future.future();
+        retriever.getConfig(ar -> {
+            if (ar.failed()) {
+                System.out.println(String.format("[ERROR] Failed to read configuration :{}", ar.cause().getMessage()));
+                configured.fail("cannot read config");
+            } else {
+                JsonObject config = ar.result();
+
+                TENANT_ID_TO_CLEAN = config.getString("iotProject");
+
+                JsonObject configInfinispan = config.getJsonObject("infinispan", new JsonObject());
+                INFINISPAN_HOST = configInfinispan.getString("host");
+                INFINISPAN_PORT = configInfinispan.getInteger("port", 11222);
+                INFINISPAN_USERNAME = configInfinispan.getString("username");
+                INFINISPAN_PASSWORD = configInfinispan.getString("password");
+                INFINISPAN_SASL_SERVER = configInfinispan.getString("saslServerName");
+                INFINISPAN_SASL_REALM = configInfinispan.getString("saslRealm");
+
+                verifyConfigurationValues();
+                configured.complete();
+            }
+        });
+
+        try {
+            configured.wait(10000);
```

Comment:
Instead of wait(), we should set handler for this future

Reply:
Don't mind, it's very silly, I was investigating strange vertx behaviour